### PR TITLE
Remove call to optimizer from the model initialization and added upda…

### PR DESCRIPTION
**Pull request by vladgets**
Removed call to optimizer from the model initialization and added update data directory function

### CODE REVIEW

File: nemo/collections/nlp/models/intent_slot_classification/intent_slot_classification_model.py [hunk 1]
```diff
@@ -108,8 +107,16 @@ def __init__(self, cfg: DictConfig, trainer: Trainer = None):
             mode='micro',
         )
 
-        # Optimizer setup needs to happen after all model weights are ready
-        self.setup_optimization(cfg.optim)
+    def update_data_dir(self, data_dir: str) -> None:
```
Comment (ekmb) [inline nemo/collections/nlp/models/intent_slot_classification/intent_slot_classification_model.py:110]: Does this solve the remaining tlt issues?

**Comment by ekmb** [comment]
@vladgets do you still want to merge this?

File: nemo/collections/nlp/models/intent_slot_classification/intent_slot_classification_model.py (hunk 1 above)
Comment (vladgets) [reply nemo/collections/nlp/models/intent_slot_classification/intent_slot_classification_model.py:110]: It solve some of the issues that I can potentially change data directory for fine tuning and evaluation. But for training I still need data set to sit inside of the model config file for the model initialization.

**Comment by vladgets** [comment]
> @vladgets do you still want to merge this?

Yes, I do

**Comment by ekmb** [comment]
> > @vladgets do you still want to merge this?
> 
> Yes, I do

Should this go to main or v1.0.0b2 @okuchaiev? It's needed to tlt

**Review by ekmb — APPROVED** [review]

**Pull request merged**